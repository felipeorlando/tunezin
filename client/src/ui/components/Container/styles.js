import styled, { css } from 'styled-components'
import PropTypes from 'prop-types'
import { measures, rules } from '../../breakpoints'

export const StyledContainer = styled.div`
  box-sizing: border-box;
  margin: 0 auto;
  padding: 0 15px;
  width: 100%;

  @media ${rules.md} {
    max-width: ${measures.md}px;
  }

  @media ${rules.lg} {
    max-width: ${measures.lg}px;
  }

  ${({ compact }) => compact && css`
    @media ${rules.md} {
      width: ${measures.md}px;
    }
  `}
`

StyledContainer.propTypes = {
  compact: PropTypes.bool
}

StyledContainer.defaultProps = {
  compact: false
}
