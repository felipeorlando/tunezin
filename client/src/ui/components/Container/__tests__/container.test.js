import React from 'react'
import renderer from 'react-test-renderer'

import Container from '..'

describe('Container', () => {
  it('renders correctly', () => {
    const component = <Container>Test</Container>
    const tree = renderer.create(component).toJSON()

    expect(tree).toMatchSnapshot()
  })

  describe('when receives compact prop', () => {
    it('renders correctly with different style', () => {
      const component = <Container compactu>Test</Container>
      const tree = renderer.create(component).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
