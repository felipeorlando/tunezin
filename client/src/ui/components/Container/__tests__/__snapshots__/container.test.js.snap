// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<div
  className="sc-bdfBwQ djeIjE"
>
  Test
</div>
`;

exports[`Container when receives compact prop renders correctly with different style 1`] = `
<div
  className="sc-bdfBwQ djeIjE"
>
  Test
</div>
`;
